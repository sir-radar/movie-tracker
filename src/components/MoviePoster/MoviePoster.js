import React from 'react';

const MoviePoster = ({image}) => {
  return (
    <div data-testid="movie-poster" className="col-12 col-md-4">
      <div className="col-12">
        <img 
          className="h-25 w-100" 
          src={ image ? `https://image.tmdb.org/t/p/w500${image}` : '/no-image.png'} alt="..."/>
      </div>
    </div>
  );
}

export default MoviePoster;
