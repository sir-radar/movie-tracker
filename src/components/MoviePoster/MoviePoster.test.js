import React from 'react';
import { render, cleanup } from '@testing-library/react';
import MoviePoster from './MoviePoster';
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer"

afterEach(cleanup)

const renderComponent = (props={}) => {
  const div = document.createElement("div");
  return render(<MoviePoster {...props}/>, div)
}

describe('MoviePoster Component', () => {

  it("renders without crashing", () => {
    renderComponent()
  })


  it("matches snapshot", () => {
    const tree = renderer.create(<MoviePoster />).toJSON();
    expect(tree).toMatchSnapshot();
  })

});