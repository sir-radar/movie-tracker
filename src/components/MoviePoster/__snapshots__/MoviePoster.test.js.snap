// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePoster Component matches snapshot 1`] = `
<div
  className="col-12 col-md-4"
  data-testid="movie-poster"
>
  <div
    className="col-12"
  >
    <img
      alt="..."
      className="h-25 w-100"
      src="/no-image.png"
    />
  </div>
</div>
`;
